pyaedt/emit_core/results/results.py

```diff
@@ -186,5 +179,44 @@ def analyze(self, revision_name=None):
                 self.current_revision = rev[0]
                 self.current_revision._load_revision()
             else:
-                print("{} not found.".format(revision_name))
+                warnings.warn("{} not found.".format(revision_name))
+        return self.current_revision
+
+    @pyaedt_function_handler()
+    def analyze(self):
+        """
+        Analyze the current revision or create a new revision if
+         the design has changed.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        rev:class:`pyaedt.modules.Revision`
+            Specified ``Revision`` object that was generated.
+
+        Examples
+        --------
+        >>> rev = aedtapp.results.analyze()
+        >>> interferers = rev.get_interferer_names()
+        >>> receivers = rev.get_receiver_names()
+        """
+        # No revisions exist, add one
+        if self.current_revision is None:
+            self.current_revision = self._add_revision()
+        # no changes since last created revision, load it
+        elif (
+            self.revisions[-1].revision_number
+            == self.emit_project.odesktop.GetActiveProject().GetActiveDesign().GetRevision()
+        ):
+            self.get_revision(self.revisions[-1].name)
+        else:
+            # there are changes since the current revision was analyzed, create
+            # a new revision
+            self.current_revision.revision_loaded = False
+            self.current_revision = self._add_revision()
+
+        # save the changes to the revision
+        self.emit_project._emit_api.save_project()
```

Comment:
@jsalant22 Is there a reason we are calling a save here?  The emit project isn't even loaded at this point is it?